ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsTable.scala

```diff
@@ -137,6 +118,37 @@ private[events] trait EventsTable {
 
     }
 
+    private def transactionTree[R](makeResponse: ApiTransactionTree => R)(
```

Comment:
Hm, seems like `transactionTree` has a built-in mechanism for what is effectively `Option#map`. Why not return `Option[ApiTransactionTree]` and let the caller do the mapping?
The caller would use it like `transactionTree(events).map(tx => GetTransactionResponse(Some(tx)))`.
Not a big deal, but would replace some "special" code with a well known concept of the scala stdlib.

Reply:
Makes sense, thanks, do you mind if I do it in the follow-up PR? It's almost ready. :slightly_smiling_face:

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5346/commits/5cef040718d07428092f4a959ae5e76027830ed4